src/Microsoft.SqlTools.ServiceLayer/ObjectManagement/ObjectTypes/Database/DatabaseHandler.cs

```diff
@@ -26,17 +82,512 @@ public override bool CanHandleType(SqlObjectType objectType)
+            {
+                if (illegalFilenameCharacters.Contains(nameChars[i]))
+                {
+                    nameChars[i] = '_';
+                }
+            }
+            return new string(nameChars);
+        }
+
+        private bool IsManagedInstance(Server server)
+        {
+            return server?.Information?.DatabaseEngineEdition == DatabaseEngineEdition.SqlManagedInstance;
+        }
+
+        private bool IsArcEnabledManagedInstance(Server server)
+        {
+            return server?.Information?.DatabaseEngineEdition == DatabaseEngineEdition.SqlAzureArcManagedInstance;
+        }
+
+        private bool IsAnyManagedInstance(Server server)
+        {
+            return (IsManagedInstance(server) || IsArcEnabledManagedInstance(server));
+        }
+
+        private string[] GetCollations(Server server, DatabasePrototype prototype, bool isNewObject)
+        {
+            var collationItems = new List<string>();
+            bool isSphinxServer = (server.VersionMajor < minimumVersionForWritableCollation);
+
+            // if we're creating a new database or this is a Sphinx Server, add "<default>" to the dropdown
+            if (isNewObject || isSphinxServer)
+            {
+                collationItems.Add(SR.general_default);
+            }
+
+            // if the server is shiloh or later, add specific collations to the dropdown
+            if (!isSphinxServer)
+            {
+                DataTable serverCollationsTable = server.EnumCollations();
+                if (serverCollationsTable != null)
+                {
+                    foreach (DataRow serverCollation in serverCollationsTable.Rows)
+                    {
+                        string collationName = (string)serverCollation["Name"];
+                        collationItems.Add(collationName);
+                    }
+                }
+            }
+
+            if (prototype.Exists)
+            {
+                System.Diagnostics.Debug.Assert(((prototype.Collation != null) && (prototype.Collation.Length != 0)),
+                    "prototype.Collation is null");
+                System.Diagnostics.Debug.Assert(collationItems.Contains(prototype.Collation),
+                    "prototype.Collation is not in the collation list");
+
+                int index = collationItems.FindIndex(collation => collation.Equals(prototype.Collation, StringComparison.InvariantCultureIgnoreCase));
+                if (index > 0)
+                {
+                    collationItems.RemoveAt(index);
+                    collationItems.Insert(0, prototype.Collation);
+                }
+            }
+            return collationItems.ToArray();
+        }
+
+        private string[] GetCollationsWithPrototypeCollation(DatabasePrototype prototype)
+        {
+            return new string[] { prototype.Collation };
+        }
+
+        private string[] GetContainmentTypes(Server server, DatabasePrototype prototype)
```

Comment:
It would be great to add comments for methods like this - both as top level xml doc and within the method itself